spec/requests/api/hello_word_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "HelloWorlds" do
+  describe "GET /public_method" do
+    before { get "/api/public_method" }
+
+    it "renders json with message" do
+      expect(response.body).to include("This method does not need authentication")
+    end
+
+    it "receives http status" do
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe "GET /private_method" do
+    let(:user) { create(:user) }
+
+    context "when user is not authenticated" do
+      before do
+        get "/api/private_method"
+      end
+
+      it "renders json with message" do
+        expect(response.body).to include("error", "Invalid token")
+      end
+
+      it "receives http status" do
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context "when user is authenticated" do
+      before do
+        post "/users/tokens/sign_in", params: { email: user.email, password: user.password }, as: :json
+        json_response = response.parsed_body
```

Comment:
I think there is a way to generate this without doing a post request, using a helper class or something. Can you double check and let me know?

If there is no way of generating it using the library, we need to create our own so we can reuse it for other methods

Reply:
Thanks for review it. I searched on devise-api's issues and I found a comment about this. Can you review again? @edimossilva

Reply:
I don't know if it's the better way to use it with a `let`. Is it better to set it up in a helper?